iceoryx2-bb/lock-free/src/mpmc/container.rs

```diff
@@ -121,10 +113,14 @@ impl<'a, T: Copy + Debug> ContainerState<'a, T> {
 #[repr(C)]
 #[derive(Debug)]
 pub struct Container<T: Copy + Debug> {
+    // must be first member, otherwise the offset calculations fail
     active_index_ptr: RelocatablePointer<AtomicBool>,
     data_ptr: RelocatablePointer<UnsafeCell<MaybeUninit<T>>>,
     capacity: usize,
     is_memory_initialized: AtomicBool,
+    container_id: UniqueId,
+    // must be the last member, since it is a relocatable container as well and then the offset
+    // calculations would again fail
     index_set: UniqueIndexSet,
```

Comment:
This is caused by the current relocatable container design. Since we cannot implement explicit move operators with rust, I had to make some sacrifices, and this is one of them.

Maybe there is a better design. But this would require multiple white board sessions and some refactoring, leading to a huge and very complex PR. For now, I would keep the comments in the code as they are and revisit this later.

Reply:
Yes, it is fine for now. I also know how to fix this ... a derive macro should do the trick